src/file_io.jl

```diff
@@ -27,25 +28,43 @@ function Base.read(io::IO, ::Type{JLSOFile})
     _versioncheck(d["metadata"]["version"], READABLE_VERSIONS)
     upgrade_jlso!(d)
     return JLSOFile(
-        d["metadata"]["version"],
-        d["metadata"]["julia"],
+        string(d["metadata"]["version"]),
+        string(d["metadata"]["julia"]),
         d["metadata"]["format"],
         d["metadata"]["compression"],
         d["metadata"]["image"],
-        d["metadata"]["pkgs"],
+        d["metadata"]["project_toml"],
+        d["metadata"]["manifest_toml"],
         d["objects"],
     )
 end
 
 function upgrade_jlso!(raw_dict::AbstractDict)
     metadata = raw_dict["metadata"]
-    if metadata["version"] ∈ semver_spec("1")
+
+    version = if isa(metadata["version"], VersionNumber)
```

Comment:
can we comment with what transitioin this is needed for?
I.e. about how v1 and v2 stored version numbers but v3 stores strings